File: php/scripts/set_cost_center_users.php
<?php
require_once(__DIR__."/../vendor/autoload.php");
require_once("utils.php");
use Namelivia\TravelPerk\ServiceProvider;
$isSandbox = false;
$travelperk = (new ServiceProvider())->build(getenv("API_KEY"), false);
$costCenter = $travelperk->costCenters()->costCenters()->get("1");
$originalUsers = array_map(function ($user) {
	return $user->id;
}, $costCenter->users);
$costCenter = $travelperk->costCenters()->costCenters()->setUsers("1")->setIds([2])->save();
write_output("set_users_for_cost_center", [
	$costCenter->id,
	$costCenter->name,
	$costCenter->archived,
	$costCenter->countUsers,
	$costCenter->users,
    $costCenter->users[0]->firstName,
    $costCenter->users[0]->lastName,
    $costCenter->users[0]->email,
    $costCenter->users[0]->id,
    $costCenter->users[0]->phone,
    $costCenter->users[0]->profilePicture,
]);
$travelperk->costCenters()->costCenters()->setUsers("1")->setIds($originalUsers)->save();

File: python/scripts/set_cost_center_users.py
import os
from travelperk_http_python.builder.builder import build
from utils import write_output
travelperk = build(os.environ["API_KEY"], False)
cost_center = travelperk.cost_centers().cost_centers().get("1")
original_users = [user.id for user in cost_center.users]
cost_center = travelperk.cost_centers().cost_centers().set_users(
    "1"
).set_ids([2]).save()
write_output("set_users_for_cost_center", [
    cost_center.id,
    cost_center.name,
    cost_center.archived,
    cost_center.count_users,
    cost_center.users,
    cost_center.users[0].first_name,
    cost_center.users[0].last_name,
    cost_center.users[0].email,
    cost_center.users[0].id,
    cost_center.users[0].phone,
    cost_center.users[0].profile_picture,
])
travelperk.cost_centers().cost_centers().set_users("1").set_ids(
    original_users
).save()
